web/src/app/core/services/project.service.ts

```diff
@@ -46,6 +54,26 @@ export class ProjectService {
       );
   }
 
+  // This function delete the project
+  showDeleteDialog(projectUid: string): void {
+    let dialogConfig: MatDialogConfig = new MatDialogConfig();
+    dialogConfig = {
+      width: '500px',
+      data: {
+        title: 'Delete Project',
+        content: 'Are you sure you want to delete?',
+      },
+    };
+    this.dialogRef = this.dialog.open(DialogConfirmationComponent, dialogConfig);
+    this.dialogRef.afterClosed()
+      .subscribe((result: boolean) => {
+        if (result === true) {
+          this.delete(projectUid)
+            .subscribe(() => this.router.navigate(['/projects']));
```

Comment:
I am not keen on nest `subscribes`.

I think
-  the condition could be moved to a `filter`
- the `delete ` moved to a `switchMap`